### ISSUE TITLE
Panic error in mostrod when it can't connect to the LN node

### ISSUE THREAD

**Issue by Catrya** [MostroP2P/mostro#345]
If the lightning node goes down, then mostrod will stop with a panic error as it cannot connect to the node

**Comment by arkanoider** [comment]
Were you able to copy the log @Catrya ?

**Comment by arkanoider** [comment]
@grunch do you remember how did you fall into this? I saw some `expect` in lightning part probably one of them is the guilty. I will tackle them in these days. If you remember the specific case where it happened write here.

**Comment by grunch** [comment]
Hey buddy, I'm going to show you one case I found, but probably this is happening in other panic calls:

1. seller creates an order
2. lightning network node shutdown
3. buyer takes the order
4. mostrod panics

```log
thread 'main' panicked at src/lightning/mod.rs:78:14:
Failed to add hold invoice: Status { code: Unavailable, message: "error trying to connect: tcp connect error: Connection refused (os error 111)", source: Some(hyper::Error(Connect, ConnectError("tcp connect error", Os { code: 111, kind: ConnectionRefused, message: "Connection refused" }))) }
```

**Comment by arkanoider** [comment]
I am tackling this kind of exceptions @grunch , what could be the desired behaviour? I mean obviously not panicking is a must, but now I am going in the direction of having a warning in `app.rs` with the ln error. Could be this is enough? Or we need to do something more to notify the ln node issue to the admin?

**Comment by grunch** [comment]
> I am tackling this kind of exceptions @grunch , what could be the desired behaviour? I mean obviously not panicking is a must, but now I am going in the direction of having a warning in `app.rs` with the ln error. Could be this is enough? Or we need to do something more to notify the ln node issue to the admin?

we can log it as fatal or critical, I mean the most critical level in the library we are using and in the fiture we can send some kind of notification to admins when those logs appears

**Comment by Catrya** [comment]
fixed via #353
